lib/rbi/rewriters/clean_tree.rb

```diff
@@ -0,0 +1,125 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RBI
+  module Rewriters
+    # Remove all definitions existing in the index from the current tree
+    #
+    # Let's create an `Index` from two different `Tree`s:
+    # ~~~rb
+    # tree1 = Parse.parse_string(<<~RBI)
+    #   class Foo
+    #     def foo; end
+    #   end
+    # RBI
+    #
+    # tree2 = Parse.parse_string(<<~RBI)
+    #   FOO = 10
+    # RBI
+    #
+    # index = Index.index(tree1, tree2)
+    # ~~~
+    #
+    # We can use `CleanTree` to remove the definitions found in the `index` from the `Tree` to clean:
+    # ~~~rb
+    # tree_to_clean = Parser.parse_string(<<~RBI)
+    #   class Foo
+    #     def foo; end
+    #     def bar; end
+    #   end
+    #   FOO = 10
+    #   BAR = 42
+    # RBI
+    #
+    # cleaned_tree, operations = CleanTree.clean(tree_to_clean)
+    #
+    # assert_equal(<<~RBI, cleaned_tree)
+    #   class Foo
+    #     def bar; end
+    #   end
+    #   BAR = 42
+    # RBI
+    #
+    # assert_equal(<<~OPERATIONS, operations.join("\n"))
+    #   Deleted ::Foo#foo at -:2:2-2-16 (duplicate from -:2:2-2:16)
+    #   Deleted ::FOO at -:5:0-5:8 (duplicate from -:1:0-1:8)
+    # OPERATIONS
+    # ~~~
+    class CleanTree < Visitor
```

Comment:
I don't love this name since "cleaning" isn't associated with "remove from index" in my mind. But I don't have a good alternative 😕. I had to read the docs in detail for it to make sense.

Reply:
Ok. I went with `RemoveKnownDefinitions`. Let me know what you think.